src/org/jgroups/fork/ForkChannel.java

```diff
@@ -271,6 +271,7 @@ protected static FORK getFORK(Channel ch, int position, Class<? extends Protocol
             if(!create_fork_if_absent)
                 throw new IllegalArgumentException("FORK not found in main stack");
             stack.insertProtocol(fork=new FORK(), position, neighbor);
+            fork.setProtocolStack(stack);
```

Comment:
@belaban on second thought it might be safer to actually construct and set the protocol stack before inserting the protocol into the stack?

```java
fork=new FORK();
fork.setProtocolStack(stack);
stack.insertProtocol(fork, position, neighbor);
```